Upgrade to phpseclib 2.0

rabbl opened the pull request.

rabbl committed:
Upgrade to phpseclib 2.0 and php7

I have upgraded phpseclib to version 2.0 and had to change some dependencies.
There were an error writing the encrypted strings to the mysql-db.
So, I have fixed this on the fast lane with an extra base64 encryption before writing and after reading.

rabbl committed:
Update travis-config adding php7 as environment

rabbl committed:
Merge master

rabbl committed:
Update DefaultControllerTest

rabbl commented:
I have updated the tests.

MichaelThessel commented on src/App/Model/CryptAES.php:
```diff
@@ -37,7 +37,7 @@ public function decrypt($crypt)
         if (!$this->cipher && !$this->setCipher()) return $crypt;
 
-        return $this->cipher->decrypt($crypt);
+        return $this->cipher->decrypt(base64_decode($crypt));
```
Why the base64 encoding? That would break backwards compatibility.

rabbl replied:
I had problems to write the encoded data to the mysql-database. So I have uses this dirty tweak. Can you please try it with your mysql-Version?

rabbl replied:
Here is the error-message: 

`General error: 1366 Incorrect string value: '\xF7\xEElVVC...' for column 'username' at row 1`

rabbl replied:
The tests in phpseclib are using bin2hex/hex2bin:
https://github.com/phpseclib/phpseclib/blob/2.0/tests/Unit/Crypt/AES/TestCase.php

MichaelThessel replied:
You db connection is probably not UTF8, have a look here. http://stackoverflow.com/a/11013986/1655741

MichaelThessel commented on src/App/Tests/Controller/DefaultControllerTest.php:
```diff
@@ -36,8 +36,7 @@ public function createApplication()
-        $client->followRedirects();
-        $crawler = $client->request('GET', '/');
+        $crawler = $client->request('GET', '/', array(), array(), array('HTTPS' => true));
 
```
Is there a specific reason why you switched the tests to run with SSL?

rabbl replied:
I had redirects to https on each call. So I switched for the tests to https.
I can remove it or change the travis-config to 

`$config['requireHttps'] = true;`

MichaelThessel replied:
Yeah it would be better to change the Travis config instead of all the tests.

rabbl committed:
Change requests to http

rabbl committed:
Remove base64 en/decryption

rabbl commented:
I have removed the base_64 encoding and it works as it should.
The config-https-option is set to false, so the requests are in the old way.

MichaelThessel commented:
Looks like some of the test requests are still done with SSL. 
- testIndex
- testApiPostCredentialsAndGetLinkToPasswordPage
- testApiPostCredentialsEmpty

rabbl committed:
Change requireHttps to true and add test

The requireHttps-flag for the travis-tests is now set to true.
I have added a test for the automatic redirect from http to https.

rabbl commented:
All tests are requesting the ssl-routes, excepting one, which is expecting the redirect to https.
